Support Docker Engine 29

mdpearson opened the pull request:
Following up on my Slack message yesterday, this PR bumps `testcontainers` to the latest version. That lets us add a `systemProperty` setting the API version to the minimum Docker API supported by the latest Docker Engine (version 29, which started rolling out a few weeks ago).

I tested this on a M4 MacBook Pro using the very latest Docker Desktop and it worked.

Fair warning: this change will prevent `testcontainers` from working with Docker Engine v24 or earlier (pre-January 2024).

mdpearson committed:
Support Docker Engine 29

ewaterman commented:
I appreciate the context on this. I think most devs aren't running tests/docker locally so the docker engine version requirement shouldn't be an issue. I've already updated locally on ubuntu and it runs fine (plus the github action tests ran successfully on this PR)

jmiranda commented:
Thanks for reviewing @ewaterman. I don't fully understand the ramifications with upgrading, so I'm fine with this as long as it doesn't break anything. 

@mdpearson Do we need to update any docs for this? Also, please create an accompanying ticket in GitHub Issues with the discussion from Slack so we have a permanent record of this issue and its history.

I'm running version 29.1.2 locally so I assume I don't have to do anything else to my development environment for this to work?
```
[jmiranda@ThinkPad-P17 ~]$ docker --version
Docker version 29.1.2, build 890dcca
```

ewaterman commented:
> Do we need to update any docs for this?

Good point. We should define the minimum version here: https://openboxes.gitbook.io/contributor-guide/release-planning/creating-a-new-docker-image (you can edit the docs via gitbook or via commit in https://github.com/openboxes/openboxes-docs/blob/main/contributor-guide/release-planning/creating-a-new-docker-image.md)

Also might be worth adding the minimum version here: https://github.com/openboxes/openboxes/blob/develop/docker/README.MD

mdpearson committed:
Add version requirements to docker/README.MD

mdpearson commented:
Hi folks, I added a small change to the gitbook.io docs (https://github.com/openboxes/openboxes-docs/pull/4), and added a change to the README.MD in this repo as well.

jmiranda commented on docker/README.MD:
```diff
@@ -15,11 +15,12 @@ The following outlines how to run a containerized instance of OpenBoxes.
 2. Navigate to the `/docker` directory
 3. Optionally create a `.env` file containing any environment variable overrides that you need. See `.env.example` for a reference. If you do not provide an override for a variable, the default will be used. For example, given a line like `{DATASOURCE_USERNAME:-openboxes}` in the compose file, if `DATASOURCE_USERNAME` is not defined in your .env file, or it is blank, a value of `openboxes` will b ...
 4. Run `docker compose up` to start your containers. You can select the compose file to use by specifying the `-f <config file>` argument. For example: `docker compose -f docker-compose-hostdb.yml up`
+5. Look in `docker-compose`'s logs for `Grails application running at http:/...../openboxes in environment: production` and connect to that URL
```
Huge Nitpick: Is "http:/...../openboxes" your way of say "http://example.com/openboxes"? 

If so, is there a more common URL for docker instances, particularly those run locally. I feel like that'll be 80% of the use cases. 

```
Grails application running at http://localhost:8080/openboxes in environment: development
```

Otherwise, I think either of the following would be slightly better.

* `http://<host>:<port>/openboxes`
* `<serverUrl>`
* `http://.../openboxes`

ewaterman replied:
The server url is defined in `docker-compose-base.yml`:
```
GRAILS_SERVER_URL: ${GRAILS_SERVER_URL:-http://localhost:8080/openboxes/}
```
so by default it'll be `http://localhost:8080/openboxes/`

jmiranda reviewed (approved):
Looks good!

ewaterman committed:
Merge branch 'develop' into new-docker-engine-support

ewaterman committed:
Update URL in Docker README instructions
